### PR TITLE
Move JDT javadoc build from ant to maven

### PR DESCRIPTION
**Pull request by laeubi**
Similar to what was done for PDE:

- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1501

### CODE REVIEW

**Comment by laeubi** [comment]
The resulting jar looks good now so this can be merged as soon as mater is open and jdt javadoc issues are fixed.

**Comment by akurtakov** [comment]
Would you please resolve the conflict so we can see where we stand with this one now?

**Comment by laeubi** [comment]
> Would you please resolve the conflict so we can see where we stand with this one now?

done.

**Comment by laeubi** [comment]
Still has javadoc issues:
- https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1582

File: eclipse-platform-parent/pom.xml [hunk 1]
```diff
@@ -687,6 +687,16 @@
                         <placement>a</placement>
                         <head>Implementation Requirements:</head>
                     </tag>
+                    <tag>
```
Comment (akurtakov) [inline eclipse-platform-parent/pom.xml:690]: Why is this needed?

Comment (laeubi) [reply eclipse-platform-parent/pom.xml:690]: JDT uses these tags on some places and otherwise javadoc would complain about unknown tags.

**Comment by laeubi** [comment]
> Is there anything still to be done here? It's good to go from my side.

There is one improvement for this regarding usage of third party sources but I'd like to do it on top of this change, so from my side it is ready to go.
